src/Files.App/Helpers/ItemListDisplayHelpers/BulkConcurrentObservableCollection.cs

```diff
@@ -131,54 +130,53 @@ protected void OnCollectionChanged(NotifyCollectionChangedEventArgs e, bool coun
             if (!isBulkOperationStarted)
             {
                 if (countChanged)
-                {
                     PropertyChanged?.Invoke(this, EventArgsCache.CountPropertyChanged);
-                }
+
                 PropertyChanged?.Invoke(this, EventArgsCache.IndexerPropertyChanged);
                 CollectionChanged?.Invoke(this, e);
             }
 
             if (IsGrouped)
             {
-                if (!(e.NewItems is null))
-                {
+                if (e.NewItems is not null)
                     AddItemsToGroup(e.NewItems.Cast<T>());
-                }
-                if (!(e.OldItems is null))
-                {
+                
+                if (e.OldItems is not null)
                     RemoveItemsFromGroup(e.OldItems.Cast<T>());
-                }
             }
         }
 
         public void ResetGroups(CancellationToken token = default)
         {
             if (!IsGrouped)
-            {
                 return;
-            }
 
             // Prevents any unwanted errors caused by bindings updating
-            GroupedCollection.ForEach(x => x.Model.PausePropertyChangedNotifications());
-            GroupedCollection.Clear();
+            GroupedCollection?.ForEach(x => x.Model.PausePropertyChangedNotifications());
+            GroupedCollection?.Clear();
             AddItemsToGroup(collection, token);
         }
 
         private void AddItemsToGroup(IEnumerable<T> items, CancellationToken token = default)
         {
+            if (GroupedCollection is null)
```

Comment:
When can `GroupedCollection` be null?

Reply:
At line @ 97 `GroupedCollection` may be assigned `null`

Reply:
Why though? 🤔

Reply:
So I did a little bit of testing to understand when it might be null and it seems that when you swap from `Home` to a folder `GroupedCollection` is assigned null
![Null](https://user-images.githubusercontent.com/102259289/190440370-f1ef0758-510b-4dae-bf70-048456a29adc.png)

Reply:
Thanks!

Reply:
Can you add a comment to the code? Would be helpful for other devs.

Reply:
Ok